modules/alignment_processing.nf

```diff
@@ -165,7 +215,15 @@ process idxstats_from_bam {
 process flagstats_from_bam {
   label 'minimap2'
 
-  publishDir "${params.output}/minimap2", mode: params.publish_dir_mode, pattern: "${bam.baseName}.flagstats.txt"
+  publishDir (
+    path: "${params.output}/intermediate",
+    mode: params.publish_dir_mode,
+    pattern: "*.sorted.flagstats.txt",
```

Comment:
This is intentional, because as you said only sorted bams are published. I don't see a point in even having unsorted bams if they all need to be sorted in the next step, but I didn't want to change too much in this branch. Honest question: am I missing something? Is there a situation where the stats could be different between the sorted and unsorted bams?

Reply:
I agree that having unsorted bams published makes no sense. 

So, I'm thinking of the general usage of this process and it would be unexpected if the result is only published when it has the suffix `.sorted`. A bam might be sorted, but might not have the suffix `.sorted` in its name.

Reply:
> So, I'm thinking of the general usage of this process and it would be unexpected if the result is only published when it has the suffix .sorted. A bam might be sorted, but might not have the suffix .sorted in its name.

Currently in the pipeline we don't have a situation where this is needed though, right? I think that when the day comes where we need that functionality then we can revisit how the pipeline decides what is published. I'm also open to suggestions though, if you have a good general solution.